pkg/beholder/message.go

```diff
@@ -222,8 +232,32 @@ func NewMetadata(attrs Attributes) *Metadata {
 	return m
 }
 
-func (m *Metadata) Validate() error {
+// validDomainAndEntityRegex allows for alphanumeric characters and ._-
+var validDomainAndEntityRegex = regexp.MustCompile(`^[a-zA-Z0-9._-]+$`)
+
+func NewMetadataValidator() *validator.Validate {
 	validate := validator.New()
+	err := validate.RegisterValidation("domain_entity", func(fl validator.FieldLevel) bool {
+		str, isStr := fl.Field().Interface().(string)
+		if !isStr {
+			return false
+		}
+		if strings.Contains(str, "__") {
+			return false
+		}
+		if !validDomainAndEntityRegex.MatchString(str) {
+			return false
+		}
+		return true
+	})
+	if err != nil {
+		panic(err)
```

Comment:
Why panic? The caller has the ability to return error.

Reply:
good point, will update

Reply:
addressed in 78b4fa890248046a0c074b712889bd44551608d3